packages/engine/src/scene/functions/loaders/VideoFunctions.test.ts

```diff
@@ -1,5 +1,287 @@
+      })
+    })
+
+    describe('Property tests for "videoSource"', () => {
+      it('should not update property', () => {
+        videoFunctions.updateVideo(entity, {})
+
+        assert(videoComponent.videoSource === sceneComponentData.videoSource)
+        assert(obj3d.userData.videoEl.src === sceneComponentData.videoSource)
+      })
+
+      it('should update property', async () => {
+        const newSource = 'fakePath.mp4'
+        videoComponent.videoSource = newSource
+        await videoFunctions.updateVideo(entity, { videoSource: newSource })
+
+        assert(obj3d.userData.videoEl.src === newSource)
+
+        videoFunctions.updateVideo(entity, { videoSource: 'newFakepath.mp4' })
+        assert(obj3d.userData.videoEl.src === newSource, 'should not update property to passed value')
+        assert(obj3d.userData.videoEl.currentTime === 1)
+        assert(obj3d.userData.mesh.material.map.encoding === sRGBEncoding)
+        assert(obj3d.userData.mesh.material.map.minFilter === LinearFilter)
+      })
+    })
+  })
+
+  describe('serializeVideo()', () => {
+    it('should properly serialize video', () => {
+      videoFunctions.deserializeVideo(entity, sceneComponent)
+      assert.deepEqual(videoFunctions.serializeVideo(entity), sceneComponent)
+    })
+
+    it('should return undefine if there is no video component', () => {
+      assert(videoFunctions.serializeVideo(entity) === undefined)
+    })
+  })
+
+  describe('parseVideoProperties()', () => {
+    it('should use default component values', () => {
+      const componentData = videoFunctions.parseVideoProperties({})
+      assert(componentData.videoSource === SCENE_COMPONENT_VIDEO_DEFAULT_VALUES.videoSource)
+      assert(componentData.elementId.includes('video-'))
+    })
+
+    it('should use passed values', () => {
+      const componentData = videoFunctions.parseVideoProperties({ ...sceneComponentData })
+      assert.deepEqual(componentData, sceneComponentData)
+    })
+  })
+
+  describe('toggleVideo()', () => {
+    it('should properly toggle video', async () => {
+      const obj3d = getComponent(entity, Object3DComponent).value
+      obj3d.userData.videoEl = new Media()
+      obj3d.userData.audioEl = new Media()
+
+      let prevStateVideo = obj3d.userData.videoEl.paused
+      let prevStateAudio = obj3d.userData.audioEl.paused
+      videoFunctions.toggleVideo(entity)
+
+      assert(obj3d.userData.videoEl.paused !== prevStateVideo)
+      assert(obj3d.userData.audioEl.paused !== prevStateAudio)
+
+      prevStateVideo = obj3d.userData.videoEl.paused
+      prevStateAudio = obj3d.userData.audioEl.paused
+      videoFunctions.toggleVideo(entity)
+
+      assert(obj3d.userData.videoEl.paused !== prevStateVideo)
+      assert(obj3d.userData.audioEl.paused !== prevStateAudio)
+    })
+
+    it('should do nothing if no audio component is there', async () => {
+      const entity = createEntity()
+      videoFunctions.toggleVideo(entity)
+      assert(true)
```

Comment:
missing assertion

Reply:
This test case is checking whether there is `videoEl` present in `object3d.userData`. If there is no `videoEl` then function should not throw any error. So a successful return from the `toggleVideo` function is in itself a assertion that this test is passed.